docs/creating-a-new-cluster.md

```diff
@@ -0,0 +1,166 @@
+--namespace=flux-system \
+--path=<the path to the flux-system folder> \
+--prune --source=flux-system \
+--validation=client flux-system
+```
+We should now have flux successully pulling and applying its initial `flux-system` kustomization.  However the other kustomizations will fail due to a lack of secrets for the cluster.
+
+#### Secrets
+
+As mentioned in [managing secrets](./managing-secrets.md) we encrypt the secrets so that they can only be decrypted in cluster by the cluster.
+
+##### PGP Key setup
+
+Run the following commands to generate a PGP key
+
+```
+mktemp -d -t .dscp-cluster-gpg
+
+GNUPGHOME=.dscp-cluster-gpg gpg \
+--quick-gen-key --batch \
+--passphrase '' --yes <cluster-name>
+```
+Export the public key as a certificate into the certs directory
+```
+mkdir certs/<cluster-name>
+
+GNUPGHOME=.dscp-cluster-gpg gpg \
+--output certs/<cluster-name>/<cluster-name>.asc \
+--export --armor <cluster-name>
+```
+Commit this and push it up to our branch.
+
+We now need to import this key into the cluster
+```
+GNUPGHOME=.dscp-cluster-gpg gpg \
+--export-secret-keys \
+--armor <cluster-name> \
+| kubectl create secret generic sops-gpg \
+--namespace=flux-system \
+--from-file=sops.asc=/dev/stdin
+```
+
+Verify that our imported key has the correct length.
+```
+$ kubectl describe secrets -n flux-system sops-gpg
+Name:         sops-gpg
+Namespace:    flux-system
+Labels:       <none>
+Annotations:  <none>
+
+Type:  Opaque
+
+Data
+====
+sops.asc:  5046 bytes
+```
+Now delete the tmp dir we used to create the key
+```
+rm -rf .dscp-cluster-gpg
+```
+
+#### Creating the genesis
+
+To bootstrap the cluster we need to create a chain genesis file along with the associated node key secrets. This can be done using the [`make-new-cluster-genesis.sh`](../scripts/make-new-cluster-genesis.sh) script. Note this script will not overwrite pre-existing node keys so before running ensure that no node keys already exist (encrypted on otherwise) in the `/clusters/<cluster_name>/secrets` folder. This folder must exist.
+
+The described script allows for generating a new cluster with any number of desired validator nodes and additional nodes for instantiation as required. For example:
+
+```
+./scripts/make-new-cluster-genesis.sh -v red -v green -v blue -a bootnode -a api-light new-cluster > new-cluster.json
+```
+
+would create a genesis for a new cluster called `new-cluster` with three validator nodes (called `red`, `green` and `blue`) along with two additional nodes (`bootnode` and `api-light`). Additional options may be specified to configure the docker image used to generate the genesis (defaults to `ghcr.io/digicatapult/vitalam-node:latest`) and the kubernetes namespace secrets should be created in (defaults to `dscp`). The script writes the final raw genesis file to stdout so can be safely redirected. This should then either be hosted publicly or built into the node to be deployed so that the chain can be referenced.
```

Comment:
I'm not sure we would want all of these secrets to be created in a single namespace.  For now I think this is ok but we should be able to specify a namespace for each `nodes` secrets.

Reply:
fair point. There's a question as to what the command line parameters should be in that case. Also should we be able to generate a genesis including existing id's to do cross org genesis stuff? There's lots of potential additions I think, this just deals with the current use case